packages/Angular/Explorer/explorer-core/src/lib/resource-browser/resource-browser.component.ts

```diff
@@ -230,32 +251,110 @@ export class ResourceBrowserComponent {
    * Refresh the component from the database based on other current state variables.
    */
   public async Refresh() {
-    try {
-      const md = new Metadata();
-      if (!this.ResourceType || !this.ResourceType.Entity)
-        throw new Error('Resource Type not set, or Resource Type does not have an Entity specified');
-
-      const rv: RunView = new RunView(); 
-      // create a combined filter for the SQL query that combines the user's provided ItemFilter, if provided, with a user filter that only includes items that are OWNED by the current user
-      // and finally filter on the current category, if one is set and we have a category entity
-      const combinedFilter = `${this.ItemFilter ? '(' + this.ItemFilter + ') AND' : ''}
-                              ([${this.UserIDFieldName}] = '${md.CurrentUser.ID}') 
-                              ${this.CurrentCategoryID && this.CategoryIDFieldName ? `AND ([${this.CategoryIDFieldName}] = '${this.CurrentCategoryID}')` : ''}`;
-      const itemResult = await rv.RunView({
-          EntityName: this.ResourceType.Entity!,
-          ExtraFilter: combinedFilter,
-          OrderBy: this.OrderBy
-        }
-      );
-      if (!itemResult || !itemResult.Success)
-        throw new Error(`Failed to load items for ${this.ResourceType.Entity}`);
+    const views: Item[] = await this.LoadViews();
+    const categories: Item[] = await this.LoadCategories();
+
+    const items = [...categories, ...views];
+    const sortedItems = this.sortItems(items);
+    this._items = sortedItems;
+  }
 
-      this._items = itemResult.Results.map(r => new Item(r, ItemType.Resource));
+  private async LoadViews(): Promise<Item[]> {
```

Comment:
@JS-BC  should this be called `LoadViews()` or something else now?

Reply:
Looking at it again, `LoadResources` is a better name for this function